pkg/pillar/zedcloud/send.go

```diff
@@ -176,6 +176,26 @@ func VerifyAllIntf(ctx *ZedCloudContext,
 		return true, remoteTemporaryFailure, intfStatusMap, nil
 	}
 
+	// For non-mgmt (i.e. app-shared) ports, the presence of a valid IP address
+	// along with DNS server and gateway is good enough for us to deem them as Success.
+	// We do not test non-mgmt ports periodically, which makes it not possible to clear
+	// any old errors on them. Here we check for presence of valid IP/DNS on non-mgmt
+	// ports and accordingly mark their status.
+	for _, port := range ctx.DeviceNetworkStatus.Ports {
+		if port.IsMgmt {
+			continue
+		}
+		foundUnicast := false
+		for _, addr := range port.AddrInfoList {
+			if !addr.Addr.IsLinkLocalUnicast() {
+				foundUnicast = true
+			}
+		}
+		if foundUnicast && len(port.DefaultRouters) > 0 && len(port.DNSServers) > 0 {
```

Comment:
I can't tell if this does the same checks as checkIfMgmtPortsHaveIPandDNS()
Would make sense to define a member function on the port (called HasIPAndDNS()) and use that in checkIfMgmtPortsHaveIPandDNS and here so that we perform the identical checks.

Reply:
checkIfMgmtPortsHaveIPandDNS is for management interface. Since we are looking for non-mgmt with valid IP DNS, I'll create the member function HasIPAndDNS for NetworkPortStatus.